core/src/main/java/org/bitcoinj/core/LegacyAddress.java

```diff
@@ -157,38 +157,31 @@ public static LegacyAddress fromScriptHash(Network network, byte[] hash160) thro
     @Deprecated
     public static LegacyAddress fromBase58(@Nullable NetworkParameters params, String base58)
             throws AddressFormatException, AddressFormatException.WrongNetwork {
-        return fromBase58( (params != null) ? params.network() : null, base58);
+        AddressParser parser = DefaultAddressParser.fromNetworks();
+        return (LegacyAddress) ((params != null)
+                ? parser.parseAddress(base58, params.network())
+                : parser.parseAddressAnyNetwork(base58));
```

Comment:
Theoretically this could directly use `parseBase58AnyNetwork()` if that currently-private method was public.

Reply:
Yeah, I thought about this, but I was thinking there might be further refactoring inside `DefaultAddressParser` and also it might be better to stick to methods defined in the `AddressParser` interface for flexibility.

I don't have a strong feeling about it one way or the other, though. We may also be able to make the  `parseBase58AnyNetwork()` and `parseBech32AnyNetwork()` methods package-private to allow them to be used in `SegwitAdress`/`LegacyAddress` without making them part of the API.

Reply:
I'll leave it up to you, just wanted to point it out.